packages/trader/src/Modules/Trading/Containers/trade-chart.tsx

```diff
@@ -0,0 +1,200 @@
+};
+
+const TradeChart = observer((props: TTradeChartProps) => {
+    const { is_accumulator, topWidgets } = props;
+    const { client, ui, common, contract_trade, portfolio } = useStore();
+    const {
+        accumulator_barriers_data,
+        accumulator_contract_barriers_data,
+        chart_type,
+        granularity,
+        markers_array,
+        has_crossed_accu_barriers,
+        updateGranularity,
+        updateChartType,
+    } = contract_trade;
+    const { all_positions } = portfolio;
+    const { is_chart_layout_default, is_chart_countdown_visible, is_dark_mode_on, is_mobile, is_positions_drawer_on } =
+        ui;
+    const { is_socket_opened, current_language } = common;
+    const { currency, is_beta_chart, should_show_eu_content } = client;
+    const {
+        chartStateChange,
+        is_trade_enabled,
+        main_barrier_flattened: main_barrier,
+        barriers_flattened: extra_barriers,
+        show_digits_stats,
+        symbol,
+        exportLayout,
+        setChartStatus,
+        chart_layout,
+        wsForget,
+        wsForgetStream,
+        wsSendRequest,
+        wsSubscribe,
+        active_symbols,
+        has_alternative_source,
+    } = useTraderStore();
+
+    const settings = {
+        countdown: is_chart_countdown_visible,
+        isHighestLowestMarkerEnabled: false, // TODO: Pending UI,
+        language: current_language.toLowerCase(),
+        position: is_chart_layout_default ? 'bottom' : 'left',
+        theme: is_dark_mode_on ? 'dark' : 'light',
+        ...(is_accumulator ? { whitespace: 190, minimumLeftBars: is_mobile ? 3 : undefined } : {}),
+    };
+
+    const { current_spot, current_spot_time } = accumulator_barriers_data || {};
+
+    const getBottomWidgets = React.useCallback(
+        ({ digits, tick }: TBottomWidgetsParams) => (
+            <ChartBottomWidgets digits={digits} tick={tick} show_accumulators_stats={is_accumulator} />
+        ),
+        [is_accumulator]
+    );
+
+    const getMarketsOrder = (active_symbols: ActiveSymbols): string[] => {
+        const synthetic_index = 'synthetic_index';
+
+        const has_synthetic_index = !!active_symbols.find(s => s.market === synthetic_index);
+        return active_symbols
+            .slice()
+            .sort((a, b) => (a.display_name < b.display_name ? -1 : 1))
+            .map(s => s.market)
+            .reduce(
+                (arr, market) => {
+                    if (arr.indexOf(market) === -1) arr.push(market);
+                    return arr;
+                },
+                has_synthetic_index ? [synthetic_index] : []
+            );
+    };
+
+    const barriers = main_barrier ? [main_barrier, ...extra_barriers] : extra_barriers;
+
+    // max ticks to display for mobile view for tick chart
+    const max_ticks = granularity === 0 ? 8 : 24;
+
+    if (!symbol || active_symbols.length === 0) return null;
+
+    return (
+        <SmartChartSwitcher
```

Comment:
SmartCharts doesn't accept any ref/innerRef and it's not a class component to forward ref without assigning it to a node. Hence, I removed unused `const charts_ref = React.useRef();` from trade.tsx, and using `SmartChartSwitcher` directly without forwarding the ref. Removed this as well:
`const SmartChartWithRef = React.forwardRef((props, ref) => <SmartChartSwitcher innerRef={ref} {...props} />);`